src/appleseed-max-impl/appleseeddisneymtl/appleseeddisneymtl.cpp

```diff
@@ -855,3 +861,8 @@ HINSTANCE AppleseedDisneyMtlClassDesc::HInstance()
 {
     return g_module;
 }
+
+bool AppleseedDisneyMtlClassDesc::IsCompatibleWithRenderer(ClassDesc& renderer_class_desc)
+{
+    return (renderer_class_desc.ClassID() == AppleseedRenderer::get_class_id()) ? true : false;
```

Comment:
My bad!! 3ds Max 2015 defines the `Class_ID` equality operator as

    int operator==(const Class_ID& cid) const;

so your code was in fact perfectly correct!